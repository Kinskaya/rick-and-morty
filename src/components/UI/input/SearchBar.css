.search-input {
    width: 40%;
    padding: 5px 35px;
    margin: 5px 0;
    border: 1px solid teal;
    border-radius: 30px;
    outline: none;
}

.search-input::placeholder {
    color: teal;
    font-style: italic;
}

form {
    display: flex;
    align-items: center;
    justify-content: center;
}

form i {
    position: absolute;
    margin-right: 385px;
    color: teal;
}
