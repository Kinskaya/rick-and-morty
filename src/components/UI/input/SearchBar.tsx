import React, { useState } from 'react';
import { useDispatch } from 'react-redux';

import { fetchData, TAPIData } from '../../../redux/cardsSlice';
import { AppDispatch } from '../../../redux/store';

import './SearchBar.css';

interface ISearch {
  addSearch(params: TAPIData): void;
}

const SearchBar: React.FunctionComponent<ISearch> = () => {
  const [value, setValue] = useState('');
  const dispatch = useDispatch<AppDispatch>();

  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    if (event.target) {
      setValue(event.target.value);
    }
  };

  const onSubmitHandler = (event: React.SyntheticEvent) => {
    event.preventDefault();
    dispatch(fetchData({ searchValue: value }));
  };
  return (
    <form onSubmit={onSubmitHandler}>
      <i className="material-icons">search</i>
      <input
        className="search-input"
        onChange={handleChange}
        type="text"
        value={value}
        placeholder="Search..."
      />
    </form>
  );
};

export default SearchBar;
