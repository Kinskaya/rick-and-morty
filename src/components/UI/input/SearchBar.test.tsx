import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import SearchBar from './SearchBar';

jest.spyOn(window.localStorage.__proto__, 'getItem');
jest.spyOn(window.localStorage.__proto__, 'setItem');

describe('SearchBar component', () => {
  it('render correctly', () => {
    const { asFragment } = render(<SearchBar addSearch={jest.fn()} />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('localStorage works correctly', () => {
    const { unmount } = render(<SearchBar addSearch={jest.fn()} />);

    expect(localStorage.getItem).toBeCalledTimes(1);
    expect(localStorage.setItem).not.toBeCalled();

    unmount();
    expect(localStorage.setItem).toBeCalledTimes(1);
  });
  it('renders search in input', () => {
    render(<SearchBar addSearch={jest.fn()} />);
    expect(screen.getByPlaceholderText('Search...')).toBeInTheDocument();
  });
});
