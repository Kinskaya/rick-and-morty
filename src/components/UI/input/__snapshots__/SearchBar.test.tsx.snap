// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar component render correctly 1`] = `
<DocumentFragment>
  <form>
    <i
      class="material-icons"
    >
      search
    </i>
    <input
      class="search-input"
      placeholder="Search..."
      type="text"
      value=""
    />
  </form>
</DocumentFragment>
`;
